### PR TITLE
James/adjust auto leader line

### PR DESCRIPTION
**Pull request by JamesMBallard**
Assign to @nandinirao. Please review.

This is adjusting how the Automatic leader line setting works. Now it will always try to default to `Bottom` and adjust from there if it won't fit in the frame. I had to switch making the vertical and horizontal adjustments so it would work in the upper corners.

This is for v.next.

cc: @ldanzinger

### CODE REVIEW

File: Imports/ArcGIS/Runtime/Toolkit/Controls.2.0/Callout.qml [hunk 1]
```diff
@@ -603,6 +605,15 @@ Item {
 
         if (leaderPosition === Enums.LeaderPosition.Automatic) {
 
+            // Move leader vertically if vertical position isn't optimal
```
Comment (nandinirao) [inline Imports/ArcGIS/Runtime/Toolkit/Controls.2.0/Callout.qml]: @JamesMBallard Last time when I tried checking vertical first, I was hitting other issues with right and left positioning not working ok. Have you tested all 8 positions?

Comment (nandinirao) [inline Imports/ArcGIS/Runtime/Toolkit/Controls.2.0/Callout.qml]: And please make sure you also try the devices.

File: Imports/ArcGIS/Runtime/Toolkit/Controls.2.0/Callout.qml [hunk 2]
```diff
@@ -296,7 +296,9 @@ Item {
 
         // set the adjustedLeaderPosition
         if (leaderPosition !== Enums.LeaderPosition.Automatic)
-            adjustedLeaderPosition = leaderPosition
+            adjustedLeaderPosition = leaderPosition;
+        else
+            adjustedLeaderPosition = Enums.LeaderPosition.Bottom;
```
Comment (nandinirao) [inline Imports/ArcGIS/Runtime/Toolkit/Controls.2.0/Callout.qml]: Starting from bottom makes sense.

File: Imports/ArcGIS/Runtime/Toolkit/Controls.2.0/Callout.qml (hunk 1 above)
Comment (JamesMBallard) [inline Imports/ArcGIS/Runtime/Toolkit/Controls.2.0/Callout.qml]: @nandinirao I tested all the positions on Android and iOS Simulator. Android looks good but there are some issues with iOS where I can't test very well. I chatted with @ldanzinger and that's a known issue right now. It was certainly no worse on iOS.

**Comment by JamesMBallard** [comment]
There is a bit of a problem on Android. I'll look into fixing that. When you move towards the edges of the screen the callout changed the leader line position way too early (edge detection calculations may be off).

**Comment by JamesMBallard** [comment]
@nandinirao I've updated the PR and retested on Windows, Android and Mac. All looks good now.

Please give it another review.

**Comment by nandinirao** [comment]
Yes, this looks good now. Thanks!

**Pull request merged**